### ISSUE TITLE
[Hunter] Serpent Sting and Chimera Shot - Serpent wrong damage

### ISSUE THREAD

**Issue by Oganku** [WoWManiaUK/Redemption#5095]
**Links:** https://www.wow-mania.com/armory/?spell=53353
https://www.wow-mania.com/armory/?spell=49001
https://www.wow-mania.com/armory/?spell=19509
Possibly related to #2770

**What is Happening:** 
1. On PTR, I could find that [Serpent Sting](https://www.wow-mania.com/armory/?spell=49001) straight up does more damage than it should for no reason (I believe its scaling double dips in ap at the moment). On a naked character, it does 6% more damage than the tooltip, but you get 5% ranged damage mod from [Ranged Weapon Specialization](https://www.wow-mania.com/armory/?spell=19509), so the discrepancy is very small. 
However, at bis gear with 100% arp and 5100 ap, Serpent Sting suddenly does 17.5% more damage than the tooltip, while it should only do +5% due to the same talent. It is therefore doing ~12% more damage than intended out of seemingly nowhere.
Testing with 4800ap and 0% arp, the increase is +16.6%, which is very close and shows the issue is NOT related to arp.

2. Currently the damage done by [Chimera Shot - Serpent](https://www.wow-mania.com/armory/?spell=53353) is too great. It seems to be scaling with something causing it to do more damage than it should/used to few weeks ago. At the moment, Chimera Shot - Serpent will do more damage than Chimera Shot in a raid environment with raid buffs and everything. With only base crit, the numbers obtained are fine, so the base calculations are good. However, with higher crit values, we see a problem with how crits are calculated for the spell. Sometimes, the damage for crits will be ~2.3x as expected, but it will also sometimes be ~4.4x, seemingly *double critting*. 
With lower gear, the quadruple damage happens far less often than with high-end gear, reinforcing the idea that the issue revolves around crit.

** Tests done on a dummy.

*It's as if [Chimera Shot - Serpent](https://www.wow-mania.com/armory/?spell=53353), is capable of critting all by itself, but it also has a chance to consider the base serpent sting damage as a crit **ON TOP** of the base crit, which can result in absurd crit damage. The problem is that serpent sting can only crit if the player is using 2 pieces of tier 9, which 99% of the players aren't. This sometimes results in critical hit levels of damage that don't actually show as crits when they only roll the *phantom crit* damage.

*TL;DR: Chimera Shot - Serpent currently behaves as though it is double rolling for crit, which will result in 3 possible damage values. 
1- The base, expected damage for the spell (fine)
2- The regular crit damage affected by the 2.3x~ multiplier (fine)
3- A *super crit* that deals double the amount of a regular crit, and more than 4 times the amount of a non-crit. (Not fine)

**What Should happen:** 
1. [Serpent Sting](https://www.wow-mania.com/armory/?spell=49001) should not strangely double dip in attack power for its scaling.

2. [Chimera Shot - Serpent](https://www.wow-mania.com/armory/?spell=53353) should only roll for crit a single time, right before damage application, and it should NEVER consider the base sting damage to be a crit if the player is not wearing 2t9.

**Both of the above issues contribute to Chimera Shot - Serpent's damage being far greater than it should**

Additional evidence:
Full ICC data for hunter damage on mania on October 30th (behaves the same for ToC and any other raid):
![image](https://user-images.githubusercontent.com/34783736/98179461-cc056780-1ecc-11eb-9757-8b4d31caba23.png)
![image](https://user-images.githubusercontent.com/34783736/98179481-da538380-1ecc-11eb-91b3-20bc532413e4.png)
We see Chimera Shot - Serpent doing more damage than Chimera Shot itself. These are from different hunters and the issue still shows.

Comparatively, here are some logs for hunters in Method for ICC clears back in retail wotlk:
![image](https://user-images.githubusercontent.com/34783736/98179662-43d39200-1ecd-11eb-9808-114f33798a1d.png)
![image](https://user-images.githubusercontent.com/34783736/98179707-5948bc00-1ecd-11eb-90af-daa56c96a27c.png)

The Serpent part of Chimera shot is only doing on average 50% of the total Chimera Shot damage, whereas on Mania it does ~15% MORE, more than twice the intended damage, likely a consequence of issues 1 and 2 above.

Sorry for the lengthy post, I just wanted to provide as much detail as necessary to facilitate fixing the issue.

**Comment by GMMetaxa** [comment]
![image](https://user-images.githubusercontent.com/34750502/98223970-26e1a200-1f53-11eb-8e72-2c019212ca53.png)
On a sidenote: Chimera Shot does 40% of the calculated total damage Serpent would do in ticks over the total amount of time. This was how it worked on retail and prolly works on Mania too. Adding the Serpent glyph adds more total damage and therefor higher Chimera damage.

**Comment by wow-mania** [comment]
Personal info is not official info. Unless official info is added we will close this one also.

**Comment by Keader** [comment]
Some more info for this:

Issue 1 - Serpent Sting when get refresh by chimera shot, stay with original bonus and it's wrong. Bonus should be recalculated everytime that it get refresh.
It make overall damage of Serpent Sting upscale, since hunters can use Serpent Sting in burst (with a lot of attack power and buffs procced) and stay with this damage in the whole fight.
It cause issues with Chimera Shot too, since Chimera Shot Serpent depends of Serpent Sting damage... so both spells get a wrong scale here.

Issue 2 - Chimera Shot Serpent is based in Serpent Sting damage. So it should not get mods, since Serpent Sting already get that. Right now Chimera Shot Serpent get mods again, duplicating some mods application

**Comment by jeyrella** [comment]
didn't we fix the SS refresh with https://github.com/WoWManiaUK/Redemption/issues/4095 though?

**Comment by Keader** [comment]
yes, ignore everything i said before.
Me and Dranah fixed both issues that i was talking about in https://github.com/WoWManiaUK/Redemption/issues/5095#issuecomment-723438839
While i was checking the code, i found a weird fucked behavior...
Someone just add crit mods again inside of Chimera Shot Serpent script.
**so right now, it's the only issue that i can confirm**
So, it's getting duplicated mods from:

SPELLMOD_CRIT_DAMAGE_BONUS
SPELL_AURA_MOD_ATTACKER_RANGED_CRIT_DAMAGE
SPELL_AURA_MOD_CRIT_DAMAGE_BONUS
SPELL_AURA_MOD_CRIT_PERCENT_VERSUS

**Comment by Oganku** [comment]
And both serpent and non-serpent can roll a crit at the same time, which makes the serpent part of the damage (double crit).

**Comment by Keader** [comment]
they are 2 independent spells.
serpent crit not means that chimera will crit. But yes, both can crit at same time, but it's correct :)

**Comment by Keader** [comment]
Please test now Oganku.
I found no issues in Serpent Sting.
Check if Chimera Shot now matches with correct damage.

**Comment by Oganku** [comment]
Still experiencing the serpent sting thing, but it's kinda negligible.

Main issue with Chimera Shot - Serpent fixed and works perfect 👌
